### DISCUSSION TITLE
Rename and Delete in Tree View

### DISCUSSION THREAD

**Discussion by dennys** [doublecmd/doublecmd#779] [General]
I try to _rename_ and _delete_ folders in Tree View, it works to delete, but it doesn't _refresh_ the tree view after the folder is deleted. (I enabled _Auto fresh_ function, but it seems only for files?)

And when I try to _rename_ a folder (use the mouse right click or press F2), it doesn't work.
I'm not sure is it a bug or should I configure anything, can anybody advise?

**Comment by alexx2000** [comment]
Tree View has a limited functionality, auto-refresh, rename and similar functions are not supported. The main Tree View task is a navigation.

**Reply by dennys** [reply]
Got it, is it possible to add some functions in Tree View?
If it's possible, I can create a feature request.

**Reply by alexx2000** [reply]
Maybe, just something simple. Rename and delete is not simple.

**Comment by goyalyashpal** [comment]
- I can see the appeal of it,
- but i would say that rather than trying to add _ALL of those_ existing functionalities to the Tree View, a better way would be to just add the tree view ability in the normal view.

So, this would be
- like the current "Command > Flat View (Ctrl + B)" which shows _all files_ for the current directory

but with some added features like
- indicating the parent directory in some way - either using slight indentation, or prefixing it before name
- the "flatness" will be on demand (rather than for all subdirs) - so to speak.

----------
A _similar_ (not same) type of thing i.e. flatness with parent indication can be seen in:
- VSC hosting sites like github when u click "Find file", 
- the code editor [helix](https://github.com/helix-editor/helix)

<img src="https://user-images.githubusercontent.com/19423063/216814888-54127008-9937-44be-b56b-d23f0c8c14ba.png" width=400px/>

<img src="https://user-images.githubusercontent.com/19423063/216814713-8d7ac149-878e-46ef-99f0-3bc48937928d.png" width=200px/>
